tests/utilities_t/test_utils.py

```diff
@@ -106,6 +107,97 @@ def test_csv_parser_slurm(self):
         )
 
 
+class TestDeepUpdate(TestCase):
+    def test_deep_update_simple_dict(self):
+        original_mapping = {"test": 123, "NoChange": 789}
+        mapping_update = {"test": 456}
+        updated_mapping = {"test": 456, "NoChange": 789}
+
+        self.assertDictEqual(
+            updated_mapping, deep_update(original_mapping, mapping_update)
+        )
+
+    def test_deep_update_nested_dict(self):
+        original_mapping = {
+            "test": {"test": 123},
+            "NoChange1": 123,
+            "NoChange2": {"Test": 789},
+        }
+        mapping_update = {"test": {"test": 123}}
+        updated_mapping = {
+            "test": {"test": 123},
+            "NoChange1": 123,
+            "NoChange2": {"Test": 789},
+        }
+
+        self.assertDictEqual(
+            updated_mapping, deep_update(original_mapping, mapping_update)
+        )
+
+    def test_deep_update_list_in_nested_dict(self):
+        original_mapping = {
+            "test": [{"test1": 123}, {"test2": 456}],
+            "test2": [1, 2, 3],
+            "NoChange1": 123,
+            "NoChange2": {"Test": 789},
+            "NoChange3": [0, 9, 8],
+        }
+
+        mapping_update = {"test": [{"test3": 789}], "test2": [4, 5, 6]}
```

Comment:
A line-wrap would be nice here, I kept misreading `test2` as being part of `test2`. The trailing comma should satisfy **black** /**flake-8** as well.
```suggestion
        mapping_update = {
            "test": [{"test3": 789}],
             "test2": [4, 5, 6],
         }
```